File: TopKFrequentElements.java
public class TopKFrequentElements {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> hm = new HashMap<>(); // hm for hashmap
        List<Integer>[] b = new List[nums.length+1]; // b for bucket!

        for (int num : nums) {
            hm.put(num, hm.getOrDefault(num, 0) + 1);
        }

        for (int key : hm.keySet()) {
            if (b[hm.get(key)] == null)
                b[hm.get(key)] = new ArrayList<Integer>();
            b[hm.get(key)].add(key);
        }

        int[] ans = new int[k];
        int counter = 0;

        for (int pos = b.length - 1; pos >= 0 && counter < k; pos--) {
            if (b[pos] != null) {
                for (int num : b[pos]) {
                    ans[counter++] = num;
                }
            }
        }
        return ans;
    }
}

File: TopKFrequentElements.py
class TopKFrequentElements:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = defaultdict(int)
        buckets = [[] for i in range(len(nums) + 1)] # must do plus 1 because the actual array 
        ans = []                                     # counts 0th index as one of the values
                                                     # ie: if len(nums)=20, then last index wood be 19
        for num in nums:
            count[num] += 1

        for key, value in count.items():
            buckets[value].append(key)
        
        for i in range(len(buckets) - 1, 0, -1):
            for n in buckets[i]:
                ans.append(n)
                if len(ans) >= k:
                    return ans
        return ans
